src/components/user/education/education.edgedb.repository.ts

```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@nestjs/common';
+import { ID, NotFoundException, PublicOf } from '~/common';
+import { e, RepoFor } from '~/core/edgedb';
+import { Education } from './dto';
+import { EducationRepository } from './education.repository';
+
+@Injectable()
+export class EducationEdgeDBRepository
+  extends RepoFor(Education, {
+    hydrate: (education) => education['*'],
+  }).withDefaults()
+  implements PublicOf<EducationRepository>
+{
+  async getUserIdByEducation(id: ID) {
+    const edu = e.cast(e.User.Education, e.uuid(id));
+    const query = e.select(e.User, (user) => ({
+      id: true,
+      filter_single: e.op(edu, 'in', user.education),
+    }));
+    const result = await this.db.run(query);
+    if (!result) {
```

Comment:
@CarsonF - I'm not sure if we want to actually throw here?  It is conceivable that not all users will have education data, so I would think that an empty set would be reasonable to return?

Reply:
This shouldn't happen since the education cannot exist without the user.
I would just wrap in an `e.assert_exists` instead of throwing.